src/terrain/clients/bags.clj

```diff
@@ -0,0 +1,47 @@
+(ns terrain.clients.bags
+  (:use [clojure-commons.error-codes]
+        [slingshot.slingshot :only [try+ throw+]])
+  (:require [clj-http.client :as http]
+            [cemerick.url :refer [url]]
+            [clojure.tools.logging :as log]
+            [terrain.util.config :as config]))
+
+(defn- bags-url
+  ([]
+   (bags-url []))
+  ([components]
+   (-> (apply url (config/bags-base-url) components)
+       (str))))
```

Comment:
`apply` will treat `nil` as an empty collection, so you can keep the same functionality without having multiple arities by doing this:

```clojure
(defn- bags-url [& [components]]
 (-> (apply url (config/bags-base-url) components)
     (str))))
```

I tend to like the brevity of the destructuring option, but this is a purely aesthetic choice. Pick whichever option you like best.

Reply:
I put this change in, thanks for suggesting it.